src/components/graph/graph.ts

```diff
@@ -244,9 +254,13 @@ export default class GraphComponent {
           }
         }, GraphComponent._SINGLE_CLICK_DELAY)
       })
-      .on('contextmenu', (ev: Event) => {
+      .on('contextmenu', (ev: IGraphNodeData) => {
+        // collapses the context menu when selecting a node to move
+        d3.select('#right-click-menu').remove()
+
         d3.event.stopPropagation()
         this._actionStream!.next(new NodeRightClickAction(ev))
+        contextMenuComponent(this._graphToSVGPosition(ev).position)
```

Comment:
create a new event called CONTEXT_MENU_OPEN_OR_SOMETHING_ALONG_THE_LINE and emit that with the position.

Then add a event handler in actions/graph.ts. See how single click is being handled